opengever/dossier/tests/test_archiver.py

```
@@ -267,3 +268,99 @@ def test_enddate_may_be_latest_dossier_end_date(self, browser):
         browser.open(self.dossier, view='transition-archive')
         self.assertEqual(date(2021, 2, 2),
                          self._get_form_date(browser, 'dossier_enddate'))
+
+
+class TestArchiveForm(IntegrationTestCase):
+
+    features = ('filing_number', )
+
+    @browsing
+    def test_resolving_and_add_filing_number(self, browser):
+        self.login(self.secretariat_user, browser)
+        browser.open(self.empty_dossier, view='transition-archive')
+
+        browser.fill({'filing prefix': 'Government',
+                      'filing Year': u'2017',
+                      'Action': 'resolve and set filing no'})
+        browser.click_on('Archive')
+
+        self.assert_workflow_state(
+            'dossier-state-resolved', self.empty_dossier)
+        self.assertEquals(['The Dossier has been resolved'], info_messages())
```

By the way, that's a good one too:
```python
statusmessages.assert_no_error_messages()
```
I usually put this after modifying something, such as creating an object.
It can help finding a problem much faster, because it fails early with the error status message.